### PR TITLE
ARROW-15431: [Python] Address docstrings in Schema

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR is adding docstring examples to:

- `pyarrow.Schema` class methods and attributes.

### CODE REVIEW

File: python/pyarrow/types.pxi [hunk 1]
```diff
@@ -1367,6 +1367,28 @@ cdef class Field(_Weakrefable):
 
 
 cdef class Schema(_Weakrefable):
+    """
+    A named collection of types a.k.a schema.
+
```
Comment (jorisvandenbossche) [inline python/pyarrow/types.pxi:1372]: Can you add a reference to `pa.schema` as the function how to create a Schema object manually? (eg with a "See Also" section)

File: python/pyarrow/types.pxi [hunk 2]
```diff
@@ -1367,6 +1367,28 @@ cdef class Field(_Weakrefable):
 
 
 cdef class Schema(_Weakrefable):
+    """
+    A named collection of types a.k.a schema.
```
Comment (jorisvandenbossche) [inline python/pyarrow/types.pxi:1371]: I would maybe try to give a bit more detailed description of what a schema is? (eg that it describes the data types of a record batch or table)

File: python/pyarrow/types.pxi [hunk 3]
```diff
@@ -1687,6 +1852,15 @@ cdef class Schema(_Weakrefable):
...
         -------
         schema: Schema
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> schema = pa.schema([
+        ...     pa.field('n_legs', pa.int64()),
+        ...     pa.field('animals', pa.string())])
+        >>> schema.remove(1)
```
Comment (jorisvandenbossche) [inline python/pyarrow/types.pxi:1862]:
```suggestion
        ...     pa.field('animals', pa.string())])

        Remove the second field of the schema:

        >>> schema.remove(1)
```

(as general feedback point: I would add a bit more this kind of "introduction sentence" briefly stating what the example does)

**Review by jorisvandenbossche — COMMENTED** [review]
Looks great! A few small points of feedback

**Comment by jorisvandenbossche** [comment]
Linting is also failing with

```
 pyarrow.lib.Schema.to_string
-> pyarrow.lib.Schema.to_string(self, truncate_metadata=True, show_field_metadata=True, show_schema_metadata=True)
PR01: Parameters {'show_field_metadata', 'show_schema_metadata'} not documented

pyarrow.lib.Schema.add_metadata
-> pyarrow.lib.Schema.add_metadata(self, metadata)
PR01: Parameters {'metadata'} not documented
```

Not fully sure why that is happening, though (since you didn't change the Parameter sections, and those actually seem to be documented)

**Comment by AlenkaF** [comment]
This linting error is really strange and also happening on another PR. Will look at it in more detail later today or tomorrow.

File: python/pyarrow/types.pxi [hunk 4]
```diff
@@ -1367,6 +1367,39 @@ cdef class Field(_Weakrefable):
 
 
 cdef class Schema(_Weakrefable):
+    """
+    A named collection of types a.k.a schema. A schema defines the
+    column names and types in a record batch or table data structure.
+
```
Comment (wjones127) [inline python/pyarrow/types.pxi:1373]:
```suggestion
    They also contain metadata about the columns. For example, schemas 
    converted from Pandas contain metadata about their original Pandas 
    types so they can be converted back to the same types.
```

**Review by wjones127 — COMMENTED** [review]
I love the example data 🦩 

Left a few minor suggestions.

File: python/pyarrow/types.pxi [hunk 5]
```diff
@@ -1367,6 +1367,41 @@ cdef class Field(_Weakrefable):
 
 
 cdef class Schema(_Weakrefable):
+    """
+    A named collection of types a.k.a schema. A schema defines the
+    column names and types in a record batch or table data structure.
+    They also contain metadata about the columns. For example, schemas 
+    converted from Pandas contain metadata about their original Pandas 
+    types so they can be converted back to the same types.
+    Warnings
```
Comment (jorisvandenbossche) [inline python/pyarrow/types.pxi:1376]:
```suggestion

    Warnings
```

File: python/pyarrow/types.pxi [hunk 6]
```diff
@@ -1446,11 +1506,43 @@ cdef class Schema(_Weakrefable):
...
+
+        >>> schema.types
+        [DataType(int64), DataType(string)]
         """
         return [field.type for field in self]
 
     @property
     def metadata(self):
+        """
+        The schema's field metadata.
```
Comment (jorisvandenbossche) [inline python/pyarrow/types.pxi:1527]:
```suggestion
        The schema's metadata.
```

? (a field can also have metadata, but here it is the schema's metadata I think

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request closed without merging**